src/shims/env.rs

```diff
@@ -20,8 +20,12 @@ impl EnvVars {
         ecx: &mut InterpCx<'mir, 'tcx, Evaluator<'tcx>>,
         mut excluded_env_vars: Vec<String>,
     ) {
-        // Exclude `TERM` var to avoid terminfo trying to open the termcap file.
-        excluded_env_vars.push("TERM".to_owned());
+
+        // FIXME: this can be removed when we have the `stat64` shim for macos.
+        if cfg!(not(target_os = "linux")) {
```

Comment:
This is doing the wrong check. You have to check if the *target* OS is Linux. Yes I know it says `target_os` but this refers to the targets OS from the perspective of the rustc that *builds* Miri, not Miri itself.

You can access the target_os like [here](https://github.com/rust-lang/miri/blob/4ebc0306ecb5a593ad9c329fa2fc636ab4a176ac/src/eval.rs#L176).